whylog/teacher/__init__.py

```diff
@@ -156,53 +173,41 @@ def set_causes_relation(self, relation):
         """
         pass
 
-    def _validate_primary_keys(self):
-        errors = []
-        for line_id in six.iterkeys(self._parsers):
-            primary_key = self._parsers[line_id].primary_key
-            pattern_match = self.pattern_assistant.get_pattern_match(line_id)
-            group_numbers = pattern_match.param_groups.keys()
-            if set(primary_key) - set(group_numbers):
-                errors.append(InvalidPrimaryKeyProblem(line_id, primary_key, group_numbers))
-        return ValidationResult(errors=errors, warnings=[])
-
-    def _validate_pattern_names(self):
-        errors = []
+    def _rule_problems(self):
+        problems = []
+        if self.effect_id is None:
+            problems.append(NoEffectParserProblem())
+        if len(self._parsers) < 2:
```

Comment:
`if len(self._parsers) == 0` this code will emit a "OneParserProblem" object. It does not seem to be correct.